finetune/base.py

```diff
@@ -146,82 +135,81 @@ def _eval(self, *tensors, feed_dict):
             if value is not None
         }
 
-    def _finetune(self, *Xs, Y=None, batch_size=None):
-        self.label_encoder = self._get_target_encoder()
+    def finetune(self, *Xs, Y=None, batch_size=None):
+        if len(Xs) > 1 and Y is None:
+            Y = Xs[-1]
+            Xs = Xs[:-1]
+        fit_language_model_only = (Y is None)
+        arr_encoded = self._text_to_ids(*Xs)
+        labels = None if fit_language_model_only else Y
+        return self._training_loop(
+            arr_encoded,
+            Y=labels,
+            batch_size=batch_size, 
+        )
 
+    def _training_loop(self, arr_encoded, Y=None, batch_size=None):
```

Comment:
This has undone half of my fix for validation split problems. This leaves the problem that at small dataset sizes the only example(s) of a given class can be split into a validation set without any errors. If you're fine with that then you can ignore this.